visualization/src/resource-visualization.cc

```diff
@@ -65,6 +65,23 @@ DEFINE_bool(
 
 namespace visualization {
 
+template <typename T_input>
+static void convertAndSubsampleIfNecessary(
+    const T_input& cloud_in, sensor_msgs::PointCloud2* ros_point_cloud) {
+  const uint32_t point_step = backend::getPointStep(
+      cloud_in.hasColor(), cloud_in.hasNormals(), cloud_in.hasScalars(),
+      cloud_in.hasLabels());
+  static const size_t kMargin = 1000u;
+  static const size_t kMaxRosMessageSize = 1000000000u;
+  static const size_t kMaxDataSize = kMaxRosMessageSize - kMargin;
```

Comment:
Just a question and a comment. Why define a margin? instead of just writing the operation there  `kMaxDataSize = 1000000000u - 1000u;`. And would it make sense to also make it a flag? Like that you could subsample also when publishing scan by scan, no?